File: ServicioGestionBancaria2/AppServicio/CuentasBancarias/Ahorros.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AppServicio.CuentasBancarias
{
    public class Ahorros : Cuentas
    {

        private static int count = 0;
        private static int numeroTransaciones = 3;


        #region "Propiedades"

        public double TasaInteres { get; set; }

        #endregion

        #region"Constructores"

        ///<summary>
        ///Contructor por defecto  de Cuenta de Ahorros
        ///</summary>
        ///
        public Ahorros() : base()
        {
            TasaInteres = 0.0;
        }

        public Ahorros(int id, int idCliente, string nombreCliente, double saldo, double tasaInteres)
                    : base(id, idCliente, nombreCliente, saldo)
        {
            TasaInteres = tasaInteres;
        }

        #endregion

        #region "Metodos Sobre Escritos"

        public override string ToString()
        {
            return base.ToString() +
                   "\nTasa de Interes: " + TasaInteres;
        }
        public override bool Equals(object obj)
        {
            Ahorros o = (Ahorros)obj;
            bool result = false;

            if ((base.Equals(o)) &&
               (this.TasaInteres == o.TasaInteres))
                result = true;

            return result;
        }

        public override int GetHashCode()
        {
            return this.ToString().GetHashCode();
        }


        public override string depositar(double valor)
        {
            Saldo += valor;
            return balance(valor, 'D');
        }
        public override string retitar(double valor)
        {
            count++;
            if (count > numeroTransaciones)
                Saldo = (Saldo - (valor + TasaInteres));

            else
                Saldo = Saldo - valor;


            return balance(valor, 'R');
        }

        public override string balance(double valor, char movimiento)
        {
            string result = "";

            switch (movimiento)
            {
                case 'R':
                    result = "\n=======Movimiento Actual======"
                    + "\nFecha: " + DateTime.Now.ToShortDateString()
                    + "\nHora: " + DateTime.Now.ToShortTimeString()
                    + "\nRetiro por Valor : " + valor
                    + "\nBalanca actual: " + Saldo;
                    break;
                case 'D':
                    result = "\n=======Movimiento Actual======"
                    + "\nFecha: " + DateTime.Now.ToShortDateString()
                    + "\nHora: " + DateTime.Now.ToShortTimeString()
                    + "\nDeposito por Valor : " + valor
                    + "\nBalanca actual: " + Saldo;
                    break;
                default:
                    result = "\n=======Movimiento Actual======"
                    + "\nFecha: " + DateTime.Now.ToShortDateString()
                    + "\nHora: " + DateTime.Now.ToShortTimeString()
                    + "\nBalanca actual: " + Saldo;
                    break;

            }


            return result;
        }
        #endregion
    }
}

File: GestionBancariaService/src/java/CuentasBancarias/Ahorros.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package CuentasBancarias;

/**
 *
 * @author Carlos
 */
public class Ahorros extends Cuentas {

    private static int count = 0;
    private static int numeroTransacciones = 3;

    private double TasaInteres;

    public double getTasaInteres() {
        return TasaInteres;
    }

    public void setTasaInteres(double TasaInteres) {
        this.TasaInteres = TasaInteres;
    }

    public Ahorros() {
        TasaInteres = 0.0;
    }

    public Ahorros(int id, int idCliente, String nombreCliente, double saldo, double tasaInteres) {
        TasaInteres = tasaInteres;
    }

    @Override
    public String depositar(double valor) {
        double a = getSaldo();
        a += valor;
        setSaldo(a);
        return balance(a, 'D');
    }

    @Override
    public String retitar(double valor) {
        count++;
        double a = getSaldo();
        double b = a;
        if (count > numeroTransacciones) {
            a = (a - (valor * TasaInteres));
        } else {
            a = a - valor;
        }
        if (a <= 0) {
            setSaldo(b);
            return "Fondos insuficientes";            
        } else {  
            setSaldo(a);
        }
        return balance(a, 'R');
    }

    @Override
    public String balance(double valor, char movimiento) {
        String resultado = "";
        switch(movimiento){
        
            case 'R':
                resultado="\n=======Movimiento Actual======"
                    + "\nFecha: No se en java como \nHora: no supe como" 
                    + "\nRetiro por Valor : " + valor
                    + "\nBalanca actual: " + getSaldo();
                    break;
            case 'D':
                resultado="\n=======Movimiento Actual======"
                    + "\nFecha: No se en java como \nHora: no supe como" 
                    + "\nReposito por Valor de : " + valor
                    + "\nBalanca actual: " + getSaldo();
                    break;
            default:
                resultado="\n=======Movimiento Actual======"
                    + "\nFecha: No se en java como \nHora: no supe como"                   
                    + "\nBalanca actual: " + getSaldo();
                    break;
        }
        return resultado;
    }

    @Override
    public String toString() {
        return super.toString() + "\nTasa de Interes: " + TasaInteres;
    }

    @Override
    public boolean equals(Object obj) {
        Ahorros o = (Ahorros) obj;
        boolean result = false;
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        if (!super.equals(obj)) {
            return false;
        } else {
            // compare subclass fields
            if (this.TasaInteres == o.TasaInteres) {
                result = true;
            }
        }
        return result;
    }

    @Override
    public int hashCode() {
        return this.toString().hashCode();
    }

}
